### PR TITLE
Receive facts gathered event

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR adds the setup to receive events and it adds the first handled event `FactsGatheredEvent`

It adds the basic setup for cloudevents and jsonschema validation that are going to be moved in https://github.com/trento-project/contracts ASAP.

The execution API now implements a behaviour, this simplify the DI and the mocking in the event handling logic so we do not rely on the GenServer/message passing while testing.

The event handling logic is responsibility of the `Wanda.Policy` module, in this PR the `GenRMQ.Consumer` implementation is not tested but it could be easily tested with an integration test while we iterate.

### CODE REVIEW

File: lib/wanda/policy.ex [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  @moduledoc """
+  Handles integration events.
+  """
+
+  alias Cloudevents.Format.V_1_0.Event, as: CloudEvent
+  alias Wanda.Execution.Fact
+
+  @facts_gathered_event "trento.checks.v1.FactsGathered"
+
+  @spec handle_event(CloudEvent.t()) :: :ok | {:error, any}
```
Comment (arbulu89) [inline lib/wanda/policy.ex:11]: Is this signature correct?
Shouldn't return `{:ok, [Fact.t()]}` or similar?

**Review by arbulu89 — APPROVED** [review]
Super nice @fabriziosestito !
Nothing really to complain about hehe

File: lib/wanda/policy.ex (hunk 1 above)
Comment (fabriziosestito) [reply lib/wanda/policy.ex:11]: nope it returns :ok because it sends a message to the genserver

File: lib/wanda/json_schema.ex [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      |> File.read!()
+      |> Jason.decode!()
+      |> ExJsonSchema.Schema.resolve()
+
+    def validate(unquote(Path.basename(schema_file, ".schema.json")), data),
+      do: ExJsonSchema.Validator.validate(unquote(Macro.escape(schema)), data)
+  end
+
+  def validate(_, _), do: {:error, :schema_not_found}
+end
```
Comment (dottorblaster) [inline lib/wanda/json_schema.ex:23]: I'm gonna print this and make some posters.

**Review by dottorblaster — APPROVED** [review]
LGTM

File: lib/wanda/json_schema.ex (hunk 2 above)
Comment (fabriziosestito) [reply lib/wanda/json_schema.ex:23]: or in the contract repo :P

**Review by nelsonkopliku — APPROVED** [review]
Ah man, really good job! :rocket:

File: lib/wanda/execution/behaviour.ex [hunk 3]
```diff
@@ -0,0 +1,10 @@
+defmodule Wanda.Execution.Behaviour do
+  @moduledoc """
+  Execution API behaviour.
+  """
+
+  alias Wanda.Execution.Fact
+
+  @callback receive_facts(execution_id :: String.t(), agent_id :: String.t(), [Fact.t()]) ::
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply lib/wanda/execution/behaviour.ex:8]: @nelsonkopliku those are for sure related to me, but the agent_id is the agent id, the target is a struct of a target execution (which also has agent_id inside) so I'm not sure they need to be squashed to the same name. I think Wanda can be agnostic about clusters but for sure agent is part of its concern.

[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply lib/wanda/execution/behaviour.ex:8]: I see, I just thought that converging toward **Target+target_id** or **Agent+agent_id** would have been more consistent. 

However, I'm completely fine with how it is.

File: lib/wanda/policy.ex [hunk 4]
```diff
@@ -0,0 +1,28 @@
+defmodule Wanda.Policy do
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply lib/wanda/policy.ex:1]: let's keep on the radar, i didn't came out with a proper name yet. maybe event handler?

[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply lib/wanda/policy.ex:1]: yep, could work

**Pull request merged**